singularity-eos/eos/eos_base.hpp

```diff
@@ -68,11 +68,11 @@ class EosBase {
                                        LambdaIndexer &&lambdas) const {
     static auto const name = SG_MEMBER_FUNC_NAME();
     static auto const cname = name.c_str();
+    CRTP copy = *(static_cast<CRTP const *>(this));
```

Comment:
good idea moving this out of the loop.

Reply:
This actually resolves #104 and comes from #105. We can't capture the `*this` pointer in a gpu kernel.